app/models/review.rb

```diff
@@ -14,13 +14,13 @@
 class Review < ApplicationRecord
 
   validates :rating, :reviewer_name, presence: true
-  validates :rating, numericality: { greater_than_or_equal_to: 0, less_than_or_equal_to: 3,
+  validates :rating, numericality: { greater_than_or_equal_to: 1, less_than_or_equal_to: 3,
                                      message: 'Rating value should be between 1 to 3' }
 
-  after_save :update_restaurant_rating, if: :rating_changed?
+  after_save :update_restaurant_rating, if: :saved_change_to_rating?
```

Comment:
this is new to me.
TIL [https://github.com/rails/rails/blob/master/activerecord%2Flib%2Factive_record%2Fattribute_methods%2Fdirty.rb#L24](link to code)